flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java

```diff
@@ -150,6 +150,14 @@ public void testMerged() {
 		validate(parameter);
 	}
 
+	@Test
+	public void testFromGenericOptionsParser() throws IOException {
+		System.setProperty("input", "myInput");
+		System.setProperty("expectedCount", "15");
+		ParameterTool parameter = ParameterTool.fromGenericOptionsParser(new String[]{"-D", "input=myInput", "-DexpectedCount=15"});
```

Comment:
Is the genericOptionsParser getting the settings from the system properties or the command line arguments?